Python/Product/TestAdapter.Executor/Pytest/TestExecutorPytest.cs

```diff
@@ -132,22 +129,46 @@ IFrameworkHandle frameworkHandle
             IFrameworkHandle frameworkHandle
         ) {
             PythonProjectSettings settings = testGroup.Key;
-            if (settings.TestFramwork != TestFrameworkType.Pytest) {
+            if (settings == null || settings.TestFramwork != TestFrameworkType.Pytest) {
                 return;
             }
 
             using (var executor = new ExecutorService(settings, frameworkHandle, runContext)) {
+                Dictionary<string, TestResult> idToResultsMap = CreatePytestIdToVsTestResultsMap(testGroup);
                 var resultsXML = executor.Run(testGroup);
 
-                var testResults = TestResultParser.Parse(resultsXML, testGroup);
-                foreach (var result in testResults) {
+                //Read pytest results from xml
+                if (File.Exists(resultsXML)) {
+                    var xmlTestResultNodes = TestResultParser.Read(resultsXML).CreateNavigator().Select("/testsuite/testcase");
+                    foreach (XPathNavigator pytestResultNode in xmlTestResultNodes) {
+                        if (_cancelRequested.WaitOne(0)) {
+                            break;
+                        }
+                        try {
+                            string pytestId = TestResultParser.GetPytestId(pytestResultNode);
+                            if (pytestId != null && idToResultsMap.TryGetValue(pytestId, out TestResult vsTestResult)) {
+                                TestResultParser.UpdateVsTestResult(vsTestResult, pytestResultNode);
+                            } else {
+                                frameworkHandle.SendMessage(TestMessageLevel.Error, Strings.ErrorTestCaseNotFound.FormatInvariant(pytestResultNode.OuterXml));
+                            }
```

Comment:
here OuterXml is something like 
<testcase classname="test_fixture_marks" file="test_fixture_marks.py" line="6" name="test_data[0]" time="0.001" />

Reply:
classname could also be something like "filenamewithoutextension.classname"
and file is relative path to project/workspace